src/main/java/net/dv8tion/jda/entities/impl/GuildImpl.java

```diff
@@ -325,6 +325,26 @@ public TextChannel getPublicChannel()
     }
 
     @Override
+    public TextChannel getTextChannelById(String id)
+    {
+        return getJDA().getTextChannelById(id) == null 
+            ? null 
+            : getTextChannels().contains(getJDA().getTextChannelById(id)) 
+                ? getJDA().getTextChannelById(id) 
+                : null;
```

Comment:
Isn't it very inefficient to lookup the channel 3 times in the global channel map ? I would store it in a local variable instead